lib/stages/visAccumulate.cpp

```diff
@@ -483,33 +477,64 @@ void visAccumulate::combine_gated(visAccumulate::internalState& gate,
 }
 
 
-void visAccumulate::finalise_output(visAccumulate::internalState& state, int freq_ind,
-                                    uint32_t total_samples) {
+void visAccumulate::finalise_output(visAccumulate::internalState& state, timespec newest_frame_time)
+{
+
     // Determine the weighting factors (if weight is zero we should just
     // multiply the visibilities by zero so as not to generate Infs)
     float w = state.sample_weight_total;
     float iw = (w != 0.0) ? (1.0 / w) : 0.0;
 
-    auto output_frame = visFrameView(state.buf, state.frame_id + freq_ind);
-
-    // Copy the visibilities into place
-    map_vis_triangle(input_remap, block_size, num_elements, freq_ind,
-                     [&](int32_t pi, int32_t bi, bool conj) {
-                         cfloat t = {(float)state.vis1[2 * bi + 1], (float)state.vis1[2 * bi]};
-                         t = !conj ? t : std::conj(t);
-                         output_frame.vis[pi] = iw * t;
-                     });
-
-    // Unpack and invert the weights
-    map_vis_triangle(input_remap, block_size, num_elements, freq_ind,
-                     [&](int32_t pi, int32_t bi, bool conj) {
-                         (void)conj;
-                         float t = state.vis2[bi];
-                         output_frame.weight[pi] = w * w / t;
-                     });
-
-    // Set the actual amount of time we accumulated for
-    output_frame.fpga_seq_total = total_samples;
+    uint32_t low_sample_cut = low_sample_fraction * num_gpu_frames * samples_per_data_set;
```

Comment:
Is there a reason you moved this calculation into the main loop?

Reply:
I didn't want to store yet another variable in the class, although I guess I could just not store `low_sample_fraction`